hw/top_earlgrey/util/opentitan_earlgrey_flash_size_reduce.py

```diff
@@ -23,46 +24,25 @@ def main():
 
     # Get path to top-level directory
     top_path = os.path.normpath(os.path.join(os.path.dirname(__file__), "../../.."))
+    top_hjson = top_path + "/hw/top_earlgrey/data/top_earlgrey.hjson"
 
-    # Change the following expressions in the following source files.
-    files = [
-        top_path + "/sw/device/lib/flash_ctrl.h",
-        top_path + "/hw/top_earlgrey/rtl/top_pkg.sv",
-        top_path + "/hw/top_earlgrey/data/top_earlgrey.hjson"
-    ]
-    match = [
-        r"(#define\s+FLASH_PAGES_PER_BANK\s+)\d+(\s*)",
-        r"(localparam\s+int\s+FLASH_PAGES_PER_BANK\s*=\s*)\d+(\s*;)",
-        r"({\s*name:\s*\"eflash\"," +
-        r"\s*clock_srcs:\s*{clk_i:\s*\"main\"}," +
-        r"\s*clock_group:\s*\"infra\"," +
-        r"\s*reset_connections:\s*{rst_ni:\s*\"lc\"}," +
-        r"\s*type:\s*\"eflash\"," +
-        r"\s*base_addr:\s*\"0x\w+\"," +
-        r"\s*swaccess:\s*\"ro\"," +
-        r"\s*size:\s*\"0x)\w+"
-    ]
-    replace = [
-        r"\g<1>32\g<2>",  # Change FLASH_PAGES_PER_BANK to 32
-        r"\g<1>32\g<2>",  # Change FLASH_PAGES_PER_BANK to 32
-        r"\g<1>10000"     # Change size to 0x10000
-    ]
-
-    # Change source files
-    for idx in range(len(files)):
-        with open(files[idx], 'r+') as file:
-            text = file.read()
-            text, num = re.subn(match[idx], replace[idx], text)
-            if num == 0:
-                print("ERROR: Cannot find regular expression " + match[idx] +
-                      " in " + files[idx] + ". "
-                      "Aborting")
-                sys.exit(1)
-            else:
-                print("Modifying " + files[idx])
-                file.seek(0)
-                file.write(text)
-                file.truncate()
+    # Modify hjson to change flash size
+    with open(top_hjson, "r") as hjson_file:
+        cfg = hjson.load(hjson_file,
+                         use_decimal=True)
+
+    # write out original version
+    # with open(top_path + "/original.hjson", "w") as hjson_file:
+    #     hjson.dump(cfg, hjson_file)
+
+    # update value
+    for mem in cfg["memory"]:
+        if mem['type'] == 'eflash':
+            mem['pages_per_bank'] = 32
+
+    # write back updated hjson
+    with open(top_hjson, "w") as hjson_file:
+        hjson.dump(cfg, hjson_file)
```

Comment:
Btw, the license header is dropped if you do this. You should probably add it back in.

Reply:
yep i've added it back.